src/handlers/history.rs

```diff
@@ -46,21 +46,45 @@ async fn handler_internal(
     body: Bytes,
 ) -> Result<Response, RpcError> {
     let project_id = query.project_id.clone();
-    let address_hash = address.clone();
-    address
-        .parse::<Address>()
-        .map_err(|_| RpcError::IdentityInvalidAddress)?;
+    let address_hash = H160::from_str(&address).map_err(|_| RpcError::IdentityInvalidAddress)?;
 
     state.validate_project_access(&project_id).await?;
     let latency_tracker_start = std::time::SystemTime::now();
-    let response = state
+    let mut response = state
         .providers
         .history_provider
-        .get_transactions(address, body, query.0)
+        .get_transactions(address_hash, body.clone(), query.0.clone())
         .await
         .tap_err(|e| {
-            error!("Failed to call transaction history with {}", e);
+            error!("Failed to call transactions history with {}", e);
         })?;
+
+    if let Some(_onramp) = query.onramp.clone() {
```

Comment:
I would prefer checking this more explicitly:

```suggestion
    if query.onramp.as_ref() == Some("coinbase") {
```